support/rag/qa-webservice/private_ai_api.py

```diff
@@ -0,0 +1,109 @@
+from fastapi import FastAPI
+from pydantic import BaseModel
+import os
+import nltk
+import psycopg2
+from nltk.corpus import stopwords
+from nltk.stem import WordNetLemmatizer
+from sentence_transformers import SentenceTransformer
+import configparser
+from pgvector.psycopg2 import register_vector
+import re
+import requests
+import json
+from openai import OpenAI
+
+# TODO: figure out how to make the parts configurable, i.e. embedding model could be configured here
+# but it would also need to be the same at the document ingestion step so that the similarity search
+# can work
+
+
+def clean_text(text):
+    """
+    TODO: Copied from the embed-ingest-job-example. Needs to be replaced by a more robust approach, something
+    off the shelf ideally.
+    """
+    text = text.lower()
+    # remove punctuation and special characters
+    text = re.sub(r"[^\w\s]", "", text)
+    # remove stopwords and lemmatize
+    stop_words = set(stopwords.words("english"))
+    lemmatizer = WordNetLemmatizer()
+    text = " ".join(
+        [lemmatizer.lemmatize(word) for word in text.split() if word not in stop_words]
+    )
+    return text
+
+
+def setup_nltk(temp_dir):
+    """
+    Set up NLTK by creating a temporary directory for NLTK data and downloading required resources.
+    """
+    from pathlib import Path 
+    nltk_data_path = Path(temp_dir) / "nltk_data"
+
+    nltk_data_path.mkdir(exist_ok=True)
+    nltk.data.path.append(str(nltk_data_path))
+    if os.path.isdir(nltk_data_path):
+        return
+
+    nltk.download("stopwords", download_dir=str(nltk_data_path))
+    nltk.download("wordnet", download_dir=str(nltk_data_path))
+
+
+class QuestionModel(BaseModel):
+    question: str
+
+
+app = FastAPI()
+
+
+@app.post("/question/")
+async def answer_question(question: QuestionModel):
+    setup_nltk(".")
+
+    config = configparser.ConfigParser()
+    config.read("api_config.ini")
+
+    # Embed the question 
+    model = SentenceTransformer("all-mpnet-base-v2")
+    embedding = model.encode(clean_text(question.question), show_progress_bar=True)
+
+    # DB connection
```

Comment:
I once got an advice. If ineed to write a comment to clarify what a section is about . It's. better to encapsulate this in a separate function with proper name and write the comment (if needed) as a function doc.